File: 227-basic-calculator-ii/227-basic-calculator-ii.cpp
class Solution {
public:
    int calculate(string s) {
        int ans = 0;
        istringstream ss("+"+s);
        int last, n;
        char op;
        while(ss >> op >> n){
            if(op == '+' || op == '-'){
                n = op == '+' ? n : -n;
                ans+=n;
            }
            
            else{
                n = op == '*' ? last*n : last/n;
                ans = ans  - last + n;
            }
            last = n;
        }
        
        return ans;
        
    }
};

File: 227-basic-calculator-ii/227-basic-calculator-ii.java
class Solution {
    public int calculate(String s) {
        Queue<Character> queue = new LinkedList<>();
        for(char c: s.toCharArray()){
            if(c != ' ')queue.offer(c);
        }
        
        queue.offer('+');
        
        char preop = '+';
        
        int last = 0, num = 0;
        
        int sum = 0;
        
        while(!queue.isEmpty()){
            char c = queue.poll();
            while(!queue.isEmpty() && (c>='0' &&  c<='9')){
                num = num*10+c-'0';
                c = queue.poll();
            }
            
            switch(preop){
                case '+':
                    sum = sum+last;
                    last = num;
                    break;
                case '-':
                    sum = sum+last;
                    last = -num;
                    break;
                case '*':
                    last*=num;
                    break;
                case '/':
                    last/=num;
                    break;
            }
            preop = c;
            num = 0;
        }
        
        return last+sum;
        
    }
}